common/src/autogluon/common/utils/deprecated.py

```diff
@@ -1,26 +1,230 @@
-import logging
-import warnings
 import functools
-from typing import Optional
+import inspect
+import warnings
+from typing import Any, Dict, Optional
+
+from packaging import version
 
+from ..version import __version__
```

Comment:
Should we add logic to be able to alter this value for testing purposes? AKA to pretend we are version `1.0.3`, etc?

Reply:
Yea. This could come in handy